File: microfrontend_android_host/app/src/main/java/com/example/microfrontendandroid/domain/usecase/LoginUseCase.kt
package com.example.microfrontendandroid.domain.usecase

import com.example.microfrontendandroid.domain.repository.AuthRepository
import javax.inject.Inject

class LoginUseCase @Inject constructor(
    private val authRepository: AuthRepository
) {
    suspend operator fun invoke(email: String, password: String, isAdmin: Boolean): Result<String> {
        return authRepository.login(email, password, isAdmin)
    }
} 

File: microfrontend_ios_host/microfrontend_ios_host/Domain/UseCases/LoginUseCase.swift
import Foundation

protocol LoginUseCase {
    func execute(email: String, password: String) async throws -> UserSession
}

struct DefaultLoginUseCase: LoginUseCase {
    private let repository: AuthRepository

    init(repository: AuthRepository) {
        self.repository = repository
    }

    func execute(email: String, password: String) async throws -> UserSession {
        try await repository.login(email: email, password: password)
    }
}
